From 2fafb1427a8e0250c97e3926064d4471402c719e Mon Sep 17 00:00:00 2001
From: dgsudharsan <sudharsand@nvidia.com>
Date: Tue, 1 Oct 2024 02:24:50 +0000
Subject: [PATCH] Increase multipath to 512


diff --git a/debian/rules b/debian/rules
index aa13106fec..2618a4a4c9 100755
--- a/debian/rules
+++ b/debian/rules
@@ -69,7 +69,7 @@ override_dh_auto_configure:
 		--disable-zeromq \
 		--enable-ospfapi \
 		--enable-bgp-vnc \
-		--enable-multipath=256 \
+		--enable-multipath=512 \
 		\
 		--enable-user=frr \
 		--enable-group=frr \
-- 
2.43.2

